### DISCUSSION TITLE
Data Profiler UI integration

### DISCUSSION THREAD

**Discussion by harshach** [open-metadata/OpenMetadata#628] [General]
Hi All,
        We are working on adding data profiler details into the UI. Here is our first iteration of the mock
 ![image](https://user-images.githubusercontent.com/38649/135485292-66c26ff6-bb1b-4d0d-bfaa-7aaf97e28e74.png)

Data profiler will run as part of the tables metadata ingestion and creates a per day profile. With this data, we can show trends and any anomalies we may have in the data.  Please do give us any feedback you may have on the UI mock.

Thanks

**Comment by juliarvalenti** [comment]
A few comments for consideration:  

- There are two date formats being used in this mockup.  The tooltip uses DD-MM-YYYY and the columns use YYYY-MM-DD.  A single format would be preferable.

![Screen Shot 2021-09-30 at 1 57 51 PM](https://user-images.githubusercontent.com/42276144/135506988-89545fac-da81-4428-9251-eecec40aad4f.png)

---

- It's unclear to me why the total_amount and total_items rows have spark lines for every value, but product_id row does not.  Is that because the product_id values are static and unchanging over the multiple profiles?  If so, for design consistency, I would recommend keeping the spark line for the "column name" row even if it is static.

![Screen Shot 2021-09-30 at 1 59 48 PM](https://user-images.githubusercontent.com/42276144/135507185-697d4615-ae4f-4270-b954-fe5b1f205326.png)

---

- The "Column Name" column switching between the actual column name and the date of the profile is quite confusing to me. 

![Screen Shot 2021-09-30 at 2 02 16 PM](https://user-images.githubusercontent.com/42276144/135507513-8a5eb0af-649e-4c46-a326-521e1363a65f.png)

**Reply by sureshms** [reply]
In one of our mockups, we had repeated the entire header again for the values in the expanded section. The header was only required because, while most column names remained the same, only the date column required a separate header. Any ideas on we can avoid repeating column names, yet incorporate dates in the expanded section?

**Reply by juliarvalenti** [reply]
Hi Suresh, 

![Screen Shot 2021-09-30 at 6 46 56 PM](https://user-images.githubusercontent.com/42276144/135539816-d5215a0c-6975-464d-ad2e-b9e5538cb48e.png)

I am considering maybe something like this to make the rows a touch more clear?  

I've added darker color for the row that is currently expanded (product_id), and a tree-like structure on the left to denote that the below rows are children of the parent.

**Reply by sureshms** [reply]
That's actually a good idea 👍.

**Reply by harshach** [reply]
+1 @juliarvalenti

**Comment by harshach** [comment]
Thanks, @juliarvalenti for the feedback. These are great suggestions.
1. Agree with using a single format
2. Yes product_id is a unique id for the products and we don't get a trend line for.
